drivers: flash: stm32 flash base address from the DTS node

FRASTM opened the pull request:
For the flash driver, the base address is the MCU internal flash address (usually 0x8000000). This PR gets  that address from the device tree node "st,stm32-nv-flash"
instead of relying on the CONFIG_FLASH_BASE_ADDRESS which might differ when building for another flash memory.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/67986

FRASTM committed:
drivers: flash: stm32 flash base address from the DTS node

For the flash driver, the base address is the MCU internal flash
address (usually 0x8000000). This PR gets the that address
from the device tree node "st,stm32-nv-flash"
instead of relying on the CONFIG_FLASH_BASE_ADDRESS
which might differ when building for another flash memory.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on drivers/flash/flash_stm32.c:
```diff
@@ -157,8 +157,7 @@ static void flash_stm32_flush_caches(const struct device *dev,
 #elif defined(CONFIG_SOC_SERIES_STM32F7X)
-	SCB_InvalidateDCache_by_Addr((uint32_t *)(CONFIG_FLASH_BASE_ADDRESS
-						  + offset), len);
+	SCB_InvalidateDCache_by_Addr((uint32_t *)(FLASH_STM32_BASE_ADDRESS
```
```suggestion
	SCB_InvalidateDCache_by_Addr((uint32_t *)(FLASH_STM32_BASE_ADDRESS + offset), len);
```

FRASTM replied:
corrected,
 (this was not detected by any build !?)

erwango reviewed (commented):
Otherwise LGTM

FRASTM committed:
tests: drivers: flash: common stm32 flash testing on nucleo_f767zi

Add the nucleo_f767zi target to the list of stm32 boards to
execute the tests/drivers/flash/common

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
adding the nucleo_f767zi target to the list of boards running the tests/drivers/flash/common/

Laczen reviewed (dismissed):
LGTM
